### PR TITLE
Add support for canonical urls in sdk

### PR DESCRIPTION
**Pull request by hackrush01**

### CODE REVIEW

File: lbrynet/schema/validator.py [hunk 1]
```diff
@@ -17,7 +17,7 @@
 
 
 def validate_claim_id(claim_id):
-    if not len(claim_id) == 40:
+    if not len(claim_id) <= 40:
```
Comment (eukreign) [inline lbrynet/schema/validator.py]: Needs to check for full 40 char length.

File: lbrynet/extras/wallet/server/db.py [hunk 2]
```diff
@@ -89,6 +96,10 @@ def flush_claims(self, batch, signed_claims_batch, outpoint_batch):
                 write_outpoint(key, claim_id)
             else:
                 delete_outpoint(key)
+        for key, claim_id in self.canonical_url_cache.items():
+            self.logger.info("HERE HERE HERE HERE {}, {}".format(key, claim_id))
```
Comment (eukreign) [inline lbrynet/extras/wallet/server/db.py]: wazzup!

File: tests/integration/test_resolve_command.py [hunk 3]
```diff
@@ -57,3 +57,92 @@ class ResolveCommand(CommandTestCase):
...
+        return account_ids
+
+    async def get_colliding_claim(self, name, channel_name, claim_id, length, account_id, claim_type):
+        while True:
+            if claim_type == "channel":
+                claim = await self.make_channel(channel_name, "0.00001", True, account_id)
+            else:
+                claim = await self.make_claim(
+                    name, "0.00001", channel_name=channel_name, confirm=True, account_id=account_id
+                )
```
Comment (eukreign) [inline tests/integration/test_resolve_command.py]: make this faster per our discussion

File: lbrynet/schema/validator.py (hunk 1 above)
Comment (hackrush01) [reply lbrynet/schema/validator.py]: I'm adding a `for_resolve` flag in this function.

**Comment by eukreign** [comment]
@hackrush01 needs to be rebased, it has conflicts.

**Comment by eukreign** [comment]
I couldn't figure out what the tests were testing and wasn't able to upgrade this code to use the new sql based wallet server. Made a new PR replacing this one: https://github.com/lbryio/lbry/pull/2173

**Pull request closed without merging**